test/models/user_test.rb

```diff
@@ -320,6 +320,47 @@ def setup_user(operation)
     assert_includes record.errors.keys, :admin
   end
 
+  test "audit of password change should be saved only once, second time audited changes should not contain password_changed" do
+    user = FactoryGirl.create(:user)
+    user = User.find_by_id(user.id) #to clear the value of user.password
+    as_admin do
+      user.password = "newpassword"
+      assert_valid user
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+      assert user.save
+      #testing after_save
+      refute user.password_changed_changed?
+      refute user.password_changed
+      refute_includes user.changed, "password_changed"
+    end
+  end
+
+  test "audit of password change should be saved" do
+    user = FactoryGirl.create(:user)
+    user = User.find_by_id(user.id) #to clear the value of user.password
+    as_admin do
+      user.password = "newpassword"
+      assert_valid user
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+    end
+  end
+
+  test "audit of password change should not be saved - due to no password change" do
```

Comment:
1. This test checks that there is _no audit_.
1. The test above ("audit of password change should be saved") doesn't assert that an audit entry is created. It only checks the return values of the new methods you added. It should verify that the audit is created.